python/cudf/cudf/core/column/column.py

```diff
@@ -894,7 +918,20 @@ def from_arrow(cls, array: pa.Array) -> ColumnBase:
             result = cls.from_pylibcudf(
                 plc.interop.from_arrow(data).columns()[0]
             )
-            # TODO: cudf_dtype_from_pa_type may be less necessary for some types
+
+            if cudf.get_option("mode.pandas_compatible"):
+                # Special handling for pandas dtypes
+                if hasattr(array, "_pandas_type"):
```

Comment:
So pyarrow arrays can have this attribute? When is that the case?

Reply:
I think this was a very early change, this is now not needed.